File: tests/data-conversion/export/export.ts
export class View {
  static count = 0;

  children: View[] = [];

  constructor() {
    View.count++;
  }

  redraw(options: {force: boolean}) {
  }
}

export class Container<T> {
  children: T[] = [];

  layout(options: {redraw: boolean}) {
  }
}

export function createView() {
  checkLeaks();
  return new View();
}

export function createContainer<T>() {
  return new Container<T>();
}

function checkLeaks() {
  if (View.count > 1000) {
  }
}

File: tests/data-conversion/export/export.cpp
#include "export.h"

// static
double View::count = 0;

View::View() {
  View::count++;
}

void View::redraw(compilets::generated::Interface1* options) {}

void View::Trace(cppgc::Visitor* visitor) const {
  compilets::TraceMember(visitor, children);
}

View::~View() = default;

View* createView() {
  checkLeaks();
  return compilets::MakeObject<View>();
}

void checkLeaks() {
  if (View::count > 1000) {}
}
